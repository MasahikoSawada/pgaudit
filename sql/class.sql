\c foo
-- DDL, logged.
CREATE TABLE a1 (col int primary key);
ALTER TABLE a1 SET (fillfactor = 100);

-- READ, logged.
SELECT * FROM a1;

-- WRITE, logged.
INSERT INTO a1 VALUES(1);
UPDATE a1 SET col = col;

-- MISC, not logged.
VACUUM a1;
REINDEX TABLE a1;

-- ROLE, logged.
CREATE USER test_user;
GRANT ALL ON a1 To test_user;

-- FUNCTION, logged.
DO $$ BEGIN EXECUTE 'select ' || '1'; END$$;
