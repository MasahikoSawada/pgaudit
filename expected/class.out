\c foo
-- DDL, logged.
CREATE TABLE a1 (col int primary key);
NOTICE:  AUDIT: SESSION,1,1,DDL,,foo,[local],pg_regress,,,,,CREATE TABLE,,,CREATE TABLE a1 (col int primary key);,<not logged>
ALTER TABLE a1 SET (fillfactor = 100);
NOTICE:  AUDIT: SESSION,2,1,DDL,,foo,[local],pg_regress,,,,,ALTER TABLE,,,ALTER TABLE a1 SET (fillfactor = 100);,<not logged>
-- READ, logged.
SELECT * FROM a1;
NOTICE:  AUDIT: SESSION,3,1,READ,,foo,[local],pg_regress,,,,,SELECT,TABLE,public.a1,SELECT * FROM a1;,<not logged>
 col 
-----
(0 rows)

-- WRITE, logged.
INSERT INTO a1 VALUES(1);
NOTICE:  AUDIT: SESSION,4,1,WRITE,,foo,[local],pg_regress,,,,,INSERT,TABLE,public.a1,INSERT INTO a1 VALUES(1);,<not logged>
UPDATE a1 SET col = col;
NOTICE:  AUDIT: SESSION,5,1,WRITE,,foo,[local],pg_regress,,,,,UPDATE,TABLE,public.a1,UPDATE a1 SET col = col;,<not logged>
-- MISC, not logged.
VACUUM a1;
REINDEX TABLE a1;
-- ROLE, logged.
CREATE USER test_user;
NOTICE:  AUDIT: SESSION,6,1,ROLE,,foo,[local],pg_regress,,,,,CREATE ROLE,,,CREATE USER test_user;,<not logged>
GRANT ALL ON a1 To test_user;
NOTICE:  AUDIT: SESSION,7,1,ROLE,,foo,[local],pg_regress,,,,,GRANT,,,GRANT ALL ON a1 To test_user;,<not logged>
-- FUNCTION, logged.
DO $$ BEGIN EXECUTE 'select ' || '* FROM a1'; END$$;
NOTICE:  AUDIT: SESSION,8,1,FUNCTION,,foo,[local],pg_regress,,,,,DO,,,DO $$ BEGIN EXECUTE 'select ' || '* FROM a1'; END$$;,<not logged>
NOTICE:  AUDIT: SESSION,8,2,READ,,foo,[local],pg_regress,,,,,SELECT,TABLE,public.a1,select * FROM a1,<not logged>
